common/plugins/eu.esdihumboldt.hale.common.schema/src/eu/esdihumboldt/hale/common/schema/model/impl/ResourceSchemaSpace.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2012 Data Harmonisation Panel
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     HUMBOLDT EU Integrated Project #030962
+ *     Data Harmonisation Panel <http://www.dhpanel.eu>
+ */
+
+package eu.esdihumboldt.hale.common.schema.model.impl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+import javax.xml.namespace.QName;
+
+import eu.esdihumboldt.hale.common.schema.model.Definition;
+import eu.esdihumboldt.hale.common.schema.model.Schema;
+import eu.esdihumboldt.hale.common.schema.model.SchemaSpace;
+import eu.esdihumboldt.hale.common.schema.model.TypeConstraint;
+import eu.esdihumboldt.hale.common.schema.model.TypeDefinition;
+import eu.esdihumboldt.hale.common.schema.model.TypeIndex;
+import eu.esdihumboldt.hale.common.schema.model.constraint.type.MappingRelevantFlag;
+
+/**
+ * Default {@link SchemaSpace} implementation with resource Id.
+ * 
+ * @author Kapil Agnihotri
+ */
+public class ResourceSchemaSpace implements SchemaSpace {
+
+	private final Map<String, Schema> schemasMap = new HashMap<String, Schema>();
+
+	private Collection<TypeDefinition> allTypes;
+
+	private Collection<TypeDefinition> mappingRelevantTypes;
+
+	/**
+	 * Adds a schema.
+	 * 
+	 * @param resourceId resource id of the schema.
+	 * 
+	 * @param schema the schema to add
+	 * @return this schema space for chaining
+	 */
+	public ResourceSchemaSpace addSchema(String resourceId, Schema schema) {
+		synchronized (this) {
+			schemasMap.put(resourceId, schema);
+			if (allTypes != null) {
+				allTypes.addAll(schema.getTypes());
+			}
+			if (mappingRelevantTypes != null) {
+				mappingRelevantTypes.addAll(schema.getMappingRelevantTypes());
+			}
+		}
+		return this;
+	}
+
+	/**
+	 * Removes a schema.
+	 * 
+	 * @param resourceId resource id of the schema.
+	 * 
+	 * @return
```

Comment:
`@return the removed schema`?

Not adding a text to the `@return` tag will cause compile warnings.

Reply:
Comment added.